### DISCUSSION TITLE
does the error object always have a response ?

### DISCUSSION THREAD

**Discussion by jmls** [sindresorhus/got#1604] [Q&A] [answered]
I am trying to write a set of unit tests , and using nock to mock the request from got. 

I am trying to test if the system catches and handles http errors, so I am using nock to reply with 404, 501 etc

With every error I throw, there always seems to be an "e.response" . However, the docs say

_Additionally, the errors may have request (Got Stream) and response (Got Response) properties depending on which phase of the request failed._

So, getting to the title of the issue - does the error object always have a response ? If not, does anyone know how to use nock to simulate that particular failure ?

**Comment by szmarczak** [comment, marked as answer]
The docs specify those who do have them and those who don't. For example: https://github.com/sindresorhus/got#gotparseerror

**Reply by rightaway** [reply]
@szmarczak https://github.com/sindresorhus/got/blob/main/documentation/8-errors.md is out of date? HTTPError doesn't say it includes `response` property. Which of those error types provide it?

**Reply by szmarczak** [reply]
Indeed, I must've forgotten about this on rewrite. PR welcome! :)

**Reply by rightaway** [reply]
Are there any others it is missing from on that page?

**Reply by szmarczak** [reply]
Feel free to check out the source code - I don't remember, I always check the source code :)

**Reply by rightaway** [reply]
@szmarczak When `response` is missing, what is the correct way to get the HTTP version that was used for the request from the RequestError object? It's not under `e.options`.

**Reply by szmarczak** [reply]
`error.request.socket.alpnProtocol` or something like this.

**Reply by rightaway** [reply]
I printed e.options and e.request and httpVersion isn't available. It is available for errors that have e.response though.
